indigo_app/templates/indigo_app/place/_work_chooser.html

```diff
@@ -79,8 +79,12 @@ <h5 class="modal-title">{% trans 'Choose a work' %}</h5>
     <div class="modal-footer">
       <a href="#" class="btn btn-link create-work" target="_blank">{% trans 'Create a new work' %}</a>
       <a href="#" class="btn btn-outline-secondary" data-bs-dismiss="modal">{% trans 'Cancel' %}</a>
-      <input type="submit" class="btn btn-primary" value="{% trans 'OK' %}" form="work-chooser-form">
+      <input type="submit" class="btn btn-primary" value="{% trans 'OK' %}" form="work-chooser-form-{{ work_field }}" formnovalidate>
     </div>
-
   </div>
 </div>
+<script>
+    document.querySelector('#work-chooser-form-{{ work_field }}').addEventListener('submit', () => {
+        $(document.querySelector('#work-chooser-modal-{{ work_field }}').parentElement).modal('hide');
+    });
+</script>
```

Comment:
@longhotsummer I've made some changes to the work chooser. It was causing some strange behaviour because of conflicting ids, Each time a modal is created from a section of the form, the work chooser form is loaded with the same set of ids causing  the form to hang. I'm using the work field to differentiate them. I tried to use vanilla JS to auto close the modal after form submit but it is not working properly. The modal dissapears but the background doesn't :cry:  I've just replace the code here like you had it in workchooser.js

Reply:
I think it may be possible to have only one work chooser modal for everything ...

Reply:
@actlikewill I think this is best solved by putting this only once into the page. So the top-level page that knows this form will be used should include the initial wrapper. All the components that then target the work chooser modal will target the same modal.

Reply:
@longhotsummer I've moved the modal into the work form. It was a a matter of changing the htmx and bootstrap targets